### ISSUE TITLE
Inverse a simple DDIMPipeline

### ISSUE THREAD

**Issue by GiilDe** [huggingface/diffusers#4120]
I want to do inversion of a simple `DDIMPipeline` using the `DDIMInverseScheduler`. Why does the `DDIMPipeline` not contain an `invert` function similarly to, for example the `StableDiffusionPix2PixZeroPipeline` class? How can I do inversion of it?

**Comment by GiilDe** [comment]
since writing this I managed to write my own `DDIMPipeline` for inversion. It is almost identical to the original one but its `__call__` function can receive and image to invert (instead of starting from random noize) and its timesteps start from 0 instead of T. It seems to be working, but I'm not sure if it really has no bugs or issues

```
class DDIMPipelineInverse(DiffusionPipeline):
    r"""
    This model inherits from [`DiffusionPipeline`]. Check the superclass documentation for the generic methods the
    library implements for all the pipelines (such as downloading or saving, running on a particular device, etc.)

    Parameters:
        unet ([`UNet2DModel`]): U-Net architecture to denoise the encoded image.
        scheduler ([`SchedulerMixin`]):
            A scheduler to be used in combination with `unet` to denoise the encoded image. Can be one of
            [`DDPMScheduler`], or [`DDIMScheduler`].
    """

    def __init__(self, unet, scheduler):
        super().__init__()

        # make sure scheduler can always be converted to DDIM
        scheduler = DDIMScheduler.from_config(scheduler.config)

        self.register_modules(unet=unet, scheduler=scheduler)

    @torch.no_grad()
    def __call__(
        self,
        batch_size: int = 1,
        image_=None,
        eta: float = 0.0,
        num_inference_steps: int = 50,
        use_clipped_model_output: Optional[bool] = None,
        output_type: Optional[str] = "pil",
        return_dict: bool = True,
    ) -> Union[ImagePipelineOutput, Tuple]:
        r"""
        Args:
            batch_size (`int`, *optional*, defaults to 1):
                The number of images to generate.
            generator (`torch.Generator`, *optional*):
                One or a list of [torch generator(s)](https://pytorch.org/docs/stable/generated/torch.Generator.html)
                to make generation deterministic.
            eta (`float`, *optional*, defaults to 0.0):
                The eta parameter which controls the scale of the variance (0 is DDIM and 1 is one type of DDPM).
            num_inference_steps (`int`, *optional*, defaults to 50):
                The number of denoising steps. More denoising steps usually lead to a higher quality image at the
                expense of slower inference.
            use_clipped_model_output (`bool`, *optional*, defaults to `None`):
                if `True` or `False`, see documentation for `DDIMScheduler.step`. If `None`, nothing is passed
                downstream to the scheduler. So use `None` for schedulers which don't support this argument.
            output_type (`str`, *optional*, defaults to `"pil"`):
                The output format of the generate image. Choose between
                [PIL](https://pillow.readthedocs.io/en/stable/): `PIL.Image.Image` or `np.array`.
            return_dict (`bool`, *optional*, defaults to `True`):
                Whether or not to return a [`~pipelines.ImagePipelineOutput`] instead of a plain tuple.

        Returns:
            [`~pipelines.ImagePipelineOutput`] or `tuple`: [`~pipelines.utils.ImagePipelineOutput`] if `return_dict` is
            True, otherwise a `tuple. When returning a tuple, the first element is a list with the generated images.
        """

        # Sample gaussian noise to begin loop
        if isinstance(self.unet.config.sample_size, int):
            image_shape = (
                batch_size,
                self.unet.config.in_channels,
                self.unet.config.sample_size,
                self.unet.config.sample_size,
            )
        else:
            image_shape = (batch_size, self.unet.config.in_channels, *self.unet.config.sample_size)

        # set step values
        self.scheduler.set_timesteps(num_inference_steps)

        image = image_

        for t in self.progress_bar(self.scheduler.timesteps[:-1]):
            # 1. predict noise model_output
            model_output = self.unet(image, t).sample

            # 2. predict previous mean of image x_t-1 and add variance depending on eta
            # eta corresponds to η in paper and should be between [0, 1]
            # do x_t -> x_t-1
            image = self.scheduler.step(
                model_output, t, image, eta=eta, use_clipped_model_output=use_clipped_model_output
            ).prev_sample

        image = (image / 2 + 0.5).clamp(0, 1)
        image = image.cpu().permute(0, 2, 3, 1).numpy()
        if output_type == "pil":
            image = self.numpy_to_pil(image)

        if not return_dict:
            return (image,)

        return ImagePipelineOutput(images=image)
```
How to use - 
```
scheduler_inv = DDIMInverseScheduler.from_config(pipeline.scheduler.config)
pipeline_inv = DDIMPipelineInverse.from_pretrained(model_id, scheduler=scheduler_inv)
pipeline_inv.scheduler = scheduler_inv
```

**Comment by patrickvonplaten** [comment]
Is this related to https://github.com/huggingface/diffusers/pull/3865 ? Also @GiilDe please don't copy-paste screenshots of code - it'd much better to copy-paste the code directly

**Comment by GiilDe** [comment]
@patrickvonplaten I don't think so. It seems that #3865 is about improvements to the scheduler. The issue that I'm raising is about the `DDIMPipeline` itself. It has no API to invert similarly to the stable diffusion pipeline which makes it inconvenient (and unclear how) to apply inversion with the basic `DDIMPipeline`.
(Sorry about the image, I changed the code so slightly that I figured a diff image would be more convenient than code)
